pyresample/geometry.py

```diff
@@ -478,6 +492,22 @@ def __init__(self, lons, lats, nprocs=1):
         elif lons.ndim > 2:
             raise ValueError('Only 1 and 2 dimensional swaths are allowed')
 
+        self.hash = None
+
+    def __hash__(self):
+        """Compute the hash of this object."""
+        if self.hash is None:
+            hasher = hashlib.sha1()
+            hasher.update(get_array_hashable(self.lons))
+            hasher.update(get_array_hashable(self.lats))
+            try:
+                hasher.update(get_array_hashable(self.lons.mask))
+            except AttributeError:
+                pass
```

Comment:
Is this really more pythonic than just saying `if hasattr(self.lons.mask)` or `isinstance(self.lons, np.ma.MaskedArray)`?

Reply:
I'm still not a fan of `try/except: pass` but if you look at the top two answers here:
https://stackoverflow.com/questions/903130/hasattr-vs-try-except-block-to-deal-with-non-existent-attributes

And the answer here:
https://stackoverflow.com/questions/21024913/python-isinstance-vs-hasattr-vs-try-except-what-is-better

I accept that this way is faster in the common case even if I think it's ugly. However, is the lons array having a `mask` attribute the common case? If not we should probably switch to `hasattr`.

Reply:
EAFP + race conditions :)

Reply:
EAFP ok. Race conditions don't really apply here.